gaitmap_mad/gaitmap_mad/event_detection/_rampp_event_detection.py

```diff
@@ -238,7 +288,160 @@ def _find_all_events(
     )
 
 
-def _detect_ic(
+def _find_all_events_for_ic_stride(
+    gyr: pd.DataFrame,
+    acc: pd.DataFrame,
+    stride_list: pd.DataFrame,
+    events: Tuple[str, ...],
+    ic_search_region: Tuple[float, float],
+    min_vel_search_win_size: int,
+    sampling_rate_hz: float,
+    gyr_ic_lowpass_filter: Optional[BaseFilter],
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """Find events in provided data by looping over single strides."""
+    gyr_ml = gyr["gyr_ml"]
+    if "ic" in events:
+        if gyr_ic_lowpass_filter is not None:
+            gyr_ml_filtered = gyr_ic_lowpass_filter.filter(
+                gyr_ml, sampling_rate_hz=sampling_rate_hz
+            ).filtered_data_.to_numpy()
+        else:
+            gyr_ml_filtered = gyr_ml.to_numpy()
+    else:
+        gyr_ml_filtered = None
+    gyr = gyr.to_numpy()
+    acc_pa = -acc["acc_pa"].to_numpy()  # have to invert acc data to work on rampp paper
+    ic_events = []
+    tc_events = []
+    min_vel_events = []
+    _is_initial_stride = False
+    for index, stride in stride_list.iterrows():
+        current_stride_start = stride["start"]
+        current_stride_end = stride["end"]
+        if "ic" in events:
```

Comment:
Just to be clear on the logic here: When we have IC strides, the idea is that we want to refine the position of the IC using the same logic that we would detect an IC in case of a segmented stride, correct?

Can't we just use the last part of the normal IC detection. For segmented strides, we use some logic to find an initial IC candidate and then use the ic-search region to refine it.

When we have a IC-to-IC stride list, I would say we already have a IC candidate. So it should be sufficient to just do the refinment using the IC-search region. This should heavily simplify the logic.

The only problem is the first stride. Here I would still check, if there is enough data available in the input to look before the stride and if not, throw a warning, and only perform the refinment in the data that is available.